tests/mocha/xml_test.js

```diff
@@ -653,53 +824,63 @@ suite('XML', function() {
-          var renderedXml = Blockly.Xml.workspaceToDom(this.renderedWorkspace);
-          Blockly.Xml.domToWorkspace(renderedXml, this.headlessWorkspace);
-          var headlessXml = Blockly.Xml.workspaceToDom(this.headlessWorkspace);
+    var assertRoundTrip = function(originWs, targetWs) {
+      var originXml = Blockly.Xml.workspaceToDom(originWs);
+      Blockly.Xml.domToWorkspace(originXml, targetWs);
+      var targetXml = Blockly.Xml.workspaceToDom(targetWs);
 
-          var renderedText = Blockly.Xml.domToText(renderedXml);
-          var headlessText = Blockly.Xml.domToText(headlessXml);
+      var expectedXmlText = Blockly.Xml.domToText(originXml);
+      var actualXmlText = Blockly.Xml.domToText(targetXml);
 
-          chai.assert.equal(headlessText, renderedText);
-        };
-      });
+      chai.assert.equal(actualXmlText, expectedXmlText);
+    };
+    suite('Rendered -> XML -> Headless -> XML', function() {
       test('Comment', function() {
         var block = Blockly.Xml.domToBlock(Blockly.Xml.textToDom(
             '<block type="empty_block"/>'
         ), this.renderedWorkspace);
         block.setCommentText('test text');
         block.getCommentIcon().setBubbleSize(100, 100);
         block.getCommentIcon().setVisible(true);
-
-        this.assertRoundTrip();
+        assertRoundTrip(this.renderedWorkspace, this.headlessWorkspace);
       });
     });
     suite('Headless -> XML -> Rendered -> XML', function() {
-      setup(function() {
-        this.assertRoundTrip = function() {
-          var headlessXml = Blockly.Xml.workspaceToDom(this.headlessWorkspace);
-          Blockly.Xml.domToWorkspace(headlessXml, this.renderedWorkspace);
-          var renderedXml = Blockly.Xml.workspaceToDom(this.renderedWorkspace);
-
-          var renderedText = Blockly.Xml.domToText(renderedXml);
-          var headlessText = Blockly.Xml.domToText(headlessXml);
-
-          chai.assert.equal(renderedText, headlessText);
-        };
-      });
       test('Comment', function() {
         var block = Blockly.Xml.domToBlock(Blockly.Xml.textToDom(
             '<block type="empty_block"/>'
         ), this.headlessWorkspace);
         block.setCommentText('test text');
         block.commentModel.size = new Blockly.utils.Size(100, 100);
         block.commentModel.pinned = true;
-
-        this.assertRoundTrip();
+        
+        assertRoundTrip(this.headlessWorkspace, this.renderedWorkspace);
       });
     });
   });
+  suite('generateVariableFieldDom', function() {
+    test('Case Sensitive', function() {
+      var varId = 'testId';
+      var type = 'testType';
+      var name = 'testName';
+
+      var mockVariableModel = {
+        type: type,
+        name: name,
+        getId: function() {
+          return varId;
+        }
+      };
+
+      var generatedXml = Blockly.Xml.domToText(
+          Blockly.Variables.generateVariableFieldDom(mockVariableModel));
+      var goldenXml =
```

Comment:
expectedXml to keep consistent with above.

Reply:
Done.